references/depth/stereo/train.py

```diff
@@ -254,12 +232,28 @@ def preprocessing(image_left, image_right, disp, valid_disp_mask):
 
     for name in val_datasets:
         val_dataset = make_dataset(name, args.dataset_root, transforms=preprocessing)
+        effective_batch_size = args.batch_size * args.world_size
+        can_do_distributed_test = effective_batch_size < len(val_dataset)
+        
+        if args.distributed and can_do_distributed_test:
+            sampler = torch.utils.data.distributed.DistributedSampler(val_dataset, shuffle=False, drop_last=False)
+        else:
+            sampler = torch.utils.data.SequentialSampler(val_dataset)
+        
+        val_loader = torch.utils.data.DataLoader(
```

Comment:
NIT, can we move `val_loader` creation to the `main` function? Similar to `train_loader` (but should be before the first evaluate)